src/dagmc.cpp

```diff
@@ -0,0 +1,406 @@
+#include "openmc/dagmc.h"
+
+#include "openmc/cell.h"
+#include "openmc/constants.h"
+#include "openmc/error.h"
+#include "openmc/file_utils.h"
+#include "openmc/geometry.h"
+#include "openmc/geometry_aux.h"
+#include "openmc/material.h"
+#include "openmc/string_utils.h"
+#include "openmc/settings.h"
+#include "openmc/surface.h"
+
+#ifdef DAGMC
+
+#include "uwuw.hpp"
+#include "dagmcmetadata.hpp"
+
+#endif
+
+#include <string>
+#include <sstream>
+#include <algorithm>
+#include <fstream>
+
+namespace openmc {
+
+#ifdef DAGMC
+const bool dagmc_enabled = true;
+#else
+const bool dagmc_enabled = false;
+#endif
+
+}
+
+#ifdef DAGMC
+
+namespace openmc {
+
+const std::string DAGMC_FILENAME = "dagmc.h5m";
+
+namespace simulation {
+
+moab::DagMC::RayHistory history;
+Direction last_dir;
+
+}
+
+namespace model {
+
+moab::DagMC* DAG;
+
+} // namespace model
+
+
+void check_dagmc_file() {
+  std::string filename = settings::path_input + DAGMC_FILENAME;
+  if (!file_exists(filename)) {
+    fatal_error("Geometry DAGMC file '" + filename + "' does not exist!");
+  }
+}
+
+bool get_uwuw_materials_xml(std::string& s) {
+  check_dagmc_file();
+  UWUW uwuw((settings::path_input + DAGMC_FILENAME).c_str());
```

Comment:
You are checking to see if the file exists on the path listed in line 57, but then you use the path listed on line 65. Consider adding a parameter in the check_dagmc_file procedure that takes the file name you want to check as a parameter.  Otherwise, you could wind up checking one file and then trying to load up another (if someone changes the path on one line and not the other).